datafusion/functions/src/string/concat.rs

```diff
@@ -261,6 +262,23 @@ impl ScalarUDFImpl for ConcatFunc {
     fn documentation(&self) -> Option<&Documentation> {
         Some(get_concat_doc())
     }
+
+    fn output_ordering(&self, inputs: &[ExprProperties]) -> Result<SortProperties> {
```

Comment:
Move that logic to fn output_preserves_lex_ordering trait impl, and remove this override

Reply:
Resolved in 8949021.